### DISCUSSION TITLE
Compile Errors in v0.49

### DISCUSSION THREAD

**Discussion by rkarikari** [jvde-github/AIS-catcher#134] [General]
Hello ....

I got the following errors related to " zip " when trying to compile the latest version 0.49 on macos Monterey.

I've never had these problems in the past.

Just wondering if it's just  with macos.

<img width="1362" alt="Screenshot 2023-07-20 at 10 36 03 PM" src="https://github.com/jvde-github/AIS-catcher/assets/115777368/198ecfac-0672-4cd4-8802-5c90947c124b">

++++ both main and single compiled ok previously  ...  both have errors now after changes made by "git pull" 

<img width="1358" alt="Screenshot 2023-07-20 at 10 37 51 PM" src="https://github.com/jvde-github/AIS-catcher/assets/115777368/27e67721-6a82-4368-b569-30da951ac6dd">

**Comment by rkarikari** [comment]
changing uppercase  ....
`Include ZIP.h `
references to lowercase 
`zip.h`    and changing the referenced filename to lowercase....
........ and removing some declarations 
   `ZIP zip;`

got me this far ...


<img width="1112" alt="Screenshot 2023-07-20 at 11 13 56 PM" src="https://github.com/jvde-github/AIS-catcher/assets/115777368/d8b42405-af13-451c-8841-6ddab73ad328">

**Comment by rkarikari** [comment]
Finally got it to work.
No mods to files were necessary whatsoever.

<img width="669" alt="Screenshot 2023-07-21 at 1 32 11 AM" src="https://github.com/jvde-github/AIS-catcher/assets/115777368/e14be7c2-3adf-4d11-999c-c9b94f0dbef1">

The culprit was  

   ` libzip  ` which was apparently installed as a dependency to another package.
The `libzip ` package contains a file  `zip.h`  which conflicts with `ZIP.h` 

Once I un-installed libzip,  AIS-catcher v0.49 installed fine as seen above.

I then reinstall the package with dependency on libzip..... and all is well.

**Reply by jvde-github** [reply]
Thanks, macos is not case sensitive. Perhaps need to look at including files in a different way to avoid conflicts. A fix might also be to include ZIP.h as follows:
```
#include "Library/ZIP.h"
```

**Reply by rkarikari** [reply]
Confirmed !

replacing ....
` #include "ZIP.h" `
with
` #include "Library/ZIP.h" `   in the files ....  IO/Netwok.h    and    IO/Server.h  resolves the conflict.

Thank you.

**Reply by jvde-github** [reply]
I installed libzip on my macos but did not get the same error. But might depend which "zip.h" file the system finds first. 

I just uploaded a patch for the above so that you don't have to apply your workaround next time.

Thanks for flagging and testing!

**Comment by jvde-github** [comment]
I propose to close the discussion as fix has been implemented.
